### PR TITLE
Fix markers removal for multiline expressions

### PR DESCRIPTION
**Pull request by mikusp**
### Pull Request Description
Second attempt on #1238 , as previously it got reverted.

### Checklist

- [x] The documentation has been updated if necessary.
- [x] The code conforms to the [Luna Style Guide](https://github.com/luna/wiki/blob/master/code-style/01.general.md) if it is Haskell.
- [x] The code has been tested where possible.

### CODE REVIEW

File: libs/luna-empire/src/Empire/Commands/Graph.hs [hunk 1]
```diff
@@ -1227,14 +1227,14 @@ prepareCopy loc@(GraphLocation _ (Breadcrumb [])) nodeIds = withUnit loc $ do
         codes <- mapM
             (\(start, len) -> Code.removeMarkers <$> Code.getAt start (start + len)) $
             zip starts lengths
-        return $ Text.intercalate "\n\n" codes
+        return $ Text.intercalate "\n" codes
```
Comment (sylwiabr) [inline libs/luna-empire/src/Empire/Commands/Graph.hs]:
```suggestion
        pure $ Text.intercalate "\n" codes
```

Comment (ljkania) [reply libs/luna-empire/src/Empire/Commands/Graph.hs]: In other places as well

File: libs/luna-empire/src/Empire/Commands/Graph.hs [hunk 2]
```diff
@@ -1244,8 +1244,8 @@ moveToOrigin metas' = map (\(MarkerNodeMeta m me) -> MarkerNodeMeta m (me & Node
                       $ map (\mnm -> FileMetadata.meta mnm ^. NodeMeta.position) metas'
 
 indent :: Int -> Text -> Text
-indent offset (Text.lines -> header:rest) =
-    Text.unlines $ header : map (\line -> Text.concat [Text.replicate offset " ", line]) rest
+indent offset (Text.lines -> lines) =
+    Text.unlines $ map (\line -> Text.concat [Text.replicate offset " ", line]) lines
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph.hs]: fmap

File: libs/luna-empire/src/Empire/Commands/Graph/Code.hs [hunk 3]
```diff
@@ -84,6 +85,12 @@ cumulativeOffsetStream tokens = scanl1 f tokens where
     f (Lexer.Token prevSpan accOffset _) (Lexer.Token span offset lexeme)
         = Lexer.Token (span+offset) (accOffset + prevSpan) lexeme
 
+indentFromOffsets :: Delta -> Delta -> Delta
+indentFromOffsets prevOffset nextOffset =
+    let eol     = "\n" :: String
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph/Code.hs]: You use it in other places too. Should we make it accessible from anywhere?

File: libs/luna-empire/src/Empire/Commands/Graph/Code.hs [hunk 4]
```diff
@@ -95,21 +102,48 @@ cumulativeOffsetStream tokens = scanl1 f tokens where
...
 --       - block start (:)
 isWrongMarker
-    :: (Lexer.Token Lexer.Symbol, Lexer.Token Lexer.Symbol)
+    :: (Delta, (Lexer.Token Lexer.Symbol, Lexer.Token Lexer.Symbol))
     -> Maybe Delta
 isWrongMarker tokens
-    | (Lexer.Token _ _ preceding,
-       Lexer.Token _ offset (Lexer.Marker _)) <- tokens =
+    | (blockIndent, (Lexer.Token _ precOffset preceding,
+       Lexer.Token _ currOffset (Lexer.Marker _))) <- tokens =
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph/Code.hs]: `...(Lexer.Marker{})...`

File: libs/luna-empire/src/Empire/Commands/Graph/Code.hs [hunk 5]
```diff
@@ -95,21 +102,48 @@ cumulativeOffsetStream tokens = scanl1 f tokens where
...
         case preceding of
+            Lexer.EOL        ->
+                let indent  = indentFromOffsets precOffset currOffset
+                in  if indent > blockIndent
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph/Code.hs]: This let is not needed. You can make it inline

**Pull request merged**